pkg/dpm/plugin.go

```diff
@@ -14,49 +14,102 @@ import (
 	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
 )
 
-type Message struct{}
-
-// DevicePluginInterface is the interface that each device plugin implementation must implement.
-// Some function are provided by DevicePlugin structure, so only those missing need to be implemented.
-type DevicePluginInterface interface {
+// PluginInterface is a mandatory interface that must be implemented by all plugins. It is
+// identical to DevicePluginServer interface of device plugin API. In version v1alpha this
+// interface contains methods Allocate and ListAndWatch. For more information see
+// https://godoc.org/k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha#DevicePluginServer
+type PluginInterface interface {
 	pluginapi.DevicePluginServer
+}
+
+// PluginInterfaceStart is an optional interface that could be implemented by plugin. If case Start
+// is implemented, it will be executed by Manager after plugin instantiation and before its
+// registartion to kubelet. This method could be used to prepare resources before they are offered
+// to Kubernetes.
+type PluginInterfaceStart interface {
 	Start() error
+}
+
+// PluginInterfaceStop is an optional interface that could be implemented by plugin. If case Stop
+// is implemented, it will be executed by Manager after the plugin is unregistered from kubelet.
+// This method could be used to tear down resources.
+type PluginInterfaceStop interface {
 	Stop() error
 }
 
 // DevicePlugin represents a gRPC server client/server.
-type DevicePlugin struct {
-	Devs         []*pluginapi.Device
-	Socket       string
-	Server       *grpc.Server
-	ResourceName string
-	Deps         DevicePluginInterface
-	Update       chan Message
-	Running      bool
-	Starting     sync.Mutex
+type devicePlugin struct {
+	DevicePlugin     PluginInterface
+	ResourceName     string
+	ResourceLastName string
```

Comment:
s/ResourceLastName/Name (to be consistent with what lister provides).

Reply:
Done